File: 백준/Silver/1049. 기타줄/기타줄.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var minSix, minOne float64
	minSix = math.MaxFloat64
	minOne = math.MaxFloat64
	var s, o float64
	var n, m int
	fmt.Fscanf(reader, "%d %d\n", &n, &m)
	for i := 0; i < m; i++ {
		fmt.Fscanf(reader, "%f %f\n", &s, &o)
		minSix = math.Min(s, minSix)
		minOne = math.Min(o, minOne)
	}

	maxDiv := math.Ceil(float64(n) / 6)
	div := n / 6
	mod := n - (n/6)*6
	res := math.Min(minOne*float64(n), math.Min(maxDiv*minSix, float64(div*int(minSix)+mod*int(minOne))))
	fmt.Printf("%0.f", res)
}

File: 백준/Silver/1049. 기타줄/기타줄.py
import math 
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
min_six=float('inf')
min_one=float('inf')
n, m = map(int, input().split())
for _ in range(m):
    s, o = map(int, input().split())
    min_six=min(s, min_six)
    min_one=min(o, min_one)
res=min(math.ceil(n/6)*min_six, min_six*(n//6)+(n-6*(n//6))*min_one, min_one*n)
print(res)
